spring-integration-mail/src/main/java/org/springframework/integration/mail/dsl/ImapIdleChannelAdapterSpec.java

```diff
@@ -0,0 +1,373 @@
+	 * @param transactionSynchronizationFactory the transactionSynchronizationFactory.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactionSynchronizationFactory(
+			TransactionSynchronizationFactory transactionSynchronizationFactory) {
+		this.target.setTransactionSynchronizationFactory(transactionSynchronizationFactory);
+		return this;
+	}
+
+	/**
+	 * Configure a chain of {@link Advice} objects for message delivery.
+	 * @param adviceChain the advice chain.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec adviceChain(Advice... adviceChain) {
+		this.adviceChain.addAll(Arrays.asList(adviceChain));
+		return this;
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@code pollingTask}.
+	 * @param transactionManager the {@link PlatformTransactionManager} to use.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactional(PlatformTransactionManager transactionManager) {
+		return transactional(transactionManager, false);
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@code pollingTask}.
+	 * @param transactionManager the {@link PlatformTransactionManager} to use.
+	 * @param handleMessageAdvice the flag to indicate the target {@link Advice} type:
+	 * {@code false} - regular {@link TransactionInterceptor}; {@code true} -
+	 * {@link org.springframework.integration.transaction.TransactionHandleMessageAdvice}
+	 * extension.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactional(PlatformTransactionManager transactionManager,
+			boolean handleMessageAdvice) {
+		return transactional(new TransactionInterceptorBuilder(handleMessageAdvice)
+				.transactionManager(transactionManager)
+				.build());
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} for the
+	 * {@code pollingTask}.
+	 * @param transactionInterceptor the {@link TransactionInterceptor} to use.
+	 * @return the spec.
+	 * @see TransactionInterceptorBuilder
+	 */
+	public ImapIdleChannelAdapterSpec transactional(TransactionInterceptor transactionInterceptor) {
+		return adviceChain(transactionInterceptor);
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with default
+	 * {@code PlatformTransactionManager} and {@link DefaultTransactionAttribute} for the
+	 * {@code pollingTask}.
+	 * @return the spec.
+	 */
+	public ImapIdleChannelAdapterSpec transactional() {
+		TransactionInterceptor transactionInterceptor = new TransactionInterceptorBuilder(false).build();
+		this.componentsToRegister.add(transactionInterceptor);
+		return transactional(transactionInterceptor);
+	}
+
+	/**
+	 * @param sendingTaskExecutor the sendingTaskExecutor.
```

Comment:
No general JavaDoc expression.
Inconsistency ? :wink: 

I know, it's existing code, but you have been nit-picking for me in my File PR :smile: